app/jobs/runtime/buildpack_installer.rb

```diff
@@ -55,6 +57,22 @@ def buildpack_uploader
           buildpack_blobstore = CloudController::DependencyLocator.instance.buildpack_blobstore
           UploadBuildpack.new(buildpack_blobstore)
         end
+
+        private
+
+        def find_existing_buildpack
+          stack = buildpack_uploader.extract_stack_from_buildpack(file)
+          return Buildpack.find(name: name, stack: stack) if stack.to_s != ''
+
+          buildpacks = Buildpack.where(name: name)
+          if buildpacks.count > 1
+            raise AmbiguousBuildpackException.new("Buildpack #{name} has #{buildpacks.count} stacks, not updated")
```

Comment:
This exception is 
1. Raised from a private method
1. Caught by the same class that raises it
1. Not re-raised

I think that this exception is only used for control flow in this class. I propose we either re-raise it (wrapped in an APIError) if it is an exceptional case or find another way of modeling the control flow without using exceptions.

Reply:
A rewording to "update failed" is more clear to me, but I leave it you to determine what is most consistent with other buildpack errors.

Reply:
@Gerg sure, we can remove the PBE. We don't want to re-raise the error since we don't want errors in buildpack installation to fail the deployment. We'd rather the operator see that a particular buildpack update failed and make fixes manually as needed.